### PR TITLE
[nan-477] hide interactive demo on enterprise and don't create example integration

### PR DESCRIPTION
**Pull request by khaliqgant**
## Describe your changes
1. Hide the interactive demo if not cloud
2. Default landing page for cloud is getting-started, otherwise is the `${env}/integrations`
3. Only allow access to getting started page if cloud
4. Only show sidebar getting-started page if cloud

## Issue ticket number and link
NAN-477

## Checklist before requesting a review (skip if just adding/editing APIs & templates)
- [ ] I added tests, otherwise the reason is: 
- [ ] I added observability, otherwise the reason is:
- [ ] I added analytics, otherwise the reason is:

### CODE REVIEW

File: packages/webapp/src/App.tsx [hunk 1]
```diff
@@ -69,7 +69,7 @@ const App = () => {
             return url.pathname;
         }
 
-        return env === 'dev' ? '/dev/getting-started' : '/prod/integrations';
+        return env === 'dev' && isCloud() ? '/dev/getting-started' : `/${env}/integrations`;
```
Comment (TBonnin) [inline packages/webapp/src/App.tsx:72]: `dev AND isCloud`. Is that even possible?

File: packages/webapp/src/components/LeftNavBar.tsx [hunk 2]
```diff
@@ -154,7 +154,7 @@ export default function LeftNavBar(props: LeftNavBarProps) {
                         </div>
                     )}
                     <div className="space-y-1">
-                        {env === 'dev' && (
+                        {isCloud() && env === 'dev' && (
```
Comment (TBonnin) [inline packages/webapp/src/components/LeftNavBar.tsx:157]: same question here

File: packages/webapp/src/App.tsx (hunk 1 above)
Comment (khaliqgant) [reply packages/webapp/src/App.tsx:72]: Dev environment 
image

on our cloud

Comment (TBonnin) [reply packages/webapp/src/App.tsx:72]: got it.

**Review by TBonnin — APPROVED** [review]

File: packages/webapp/src/components/LeftNavBar.tsx (hunk 2 above)
Comment (bodinsamuel) [reply packages/webapp/src/components/LeftNavBar.tsx:157]: I didn't reviewed this sorry, but that means we can't access the getting started locally. It's probably okay but I'm not sure it was the goal :D

Comment (khaliqgant) [reply packages/webapp/src/components/LeftNavBar.tsx:157]: No worries. It wasn't the goal but was necessary to properly lock the getting-started page down. The only reason it was so permissive before was to make it easy to work on locally. Locally already as it is we can't access the team page without editing some code in the `App.tsx` so I felt this was ok

**Pull request merged**